picoquic/util.c

```diff
@@ -366,6 +366,26 @@ void picoquic_set64_connection_id(picoquic_connection_id_t * cnx_id, uint64_t va
     cnx_id->id_len = 8;
 }
 
+uint64_t picoquic_hash_addr(const struct sockaddr* addr)
+{
+    uint64_t h;
+
+    if (addr->sa_family == AF_INET) {
+        struct sockaddr_in* a4 = (struct sockaddr_in*)addr;
+        h = picohash_bytes((uint8_t*)&a4->sin_addr , 4);
+        h += 128ull * a4->sin_port;
+        h ^= 0x04040404;
+    }
+    else {
+        struct sockaddr_in6* a6 = (struct sockaddr_in6*)addr;
+        h = picohash_bytes((uint8_t*)& a6->sin6_addr, 4);
```

Comment:
Copy & paste? 4 should be 16 or some "sizeof"

Reply:
Good catch!